src/sqlfluff/rules/L064.py

```diff
@@ -0,0 +1,262 @@
+"""Implementation of Rule L064."""
+
+import sys
+from typing import Optional
+
+import regex
+
+if sys.version_info < (3, 8):
+    from typing_extensions import Final
+else:
+    from typing import Final
+
+from sqlfluff.core.parser.segments.raw import CodeSegment
+from sqlfluff.core.rules.base import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.core.rules.doc_decorators import (
+    document_configuration,
+    document_fix_compatible,
+)
+
+QUOTES_MAPPING: Final = {
+    "single_quotes": {
+        "common_name": "single quotes",
+        "preferred_quote_char": "'",
+        "alternate_quote_char": '"',
+    },
+    "double_quotes": {
+        "common_name": "double quotes",
+        "preferred_quote_char": '"',
+        "alternate_quote_char": "'",
+    },
+}
+# BigQuery string prefix characters.
+STRING_PREFIX_CHARS: Final = "rbRB"
+
+
+@document_configuration
+@document_fix_compatible
+class Rule_L064(BaseRule):
+    r"""Consistent usage of preferred quotes for STRING datatype.
+
+    Some databases allow single quotes as well as double quotes to express a STRING
+    literal. Prefer one type of quotes as specified in rule setting. Falling back to
+    alternate quotes to reduce escapes.
+
+    Dollar quoted raw strings are excluded from this rule, as they are mostly used for
+    literal UDF Body definitions.
+
+    .. note::
+       This rule can be dangerous. If accidentally enabled for dialects that do not
+       support single *and* double quotes automated fixes can potentially break working
+       SQL code.
+
+       This rule is disabled by default for dialects that don't allow double quotes for
+       STRING datatype (e.g. Postgres). It can be enabled with the
+       ``force_enable = True`` flag.
```

Comment:
```suggestion
    r"""Consistent usage of preferred quotes for quoted literals.

    Some databases allow single quotes as well as double quotes to express a quoted
    literal. Prefer one type of quotes as specified in rule setting. Falling back to
    alternate quotes to reduce escapes.

    Dollar quoted raw strings are excluded from this rule, as they are mostly used for
    literal UDF Body definitions.

    .. note::
       This rule only checks quoted literals and not quoted identifiers as they often
       cannot interchange single and double quotes

       This rule can be dangerous. If accidentally enabled for dialects that do not
       support single *and* double quotes automated fixes can potentially break working
       SQL code.

       This rule is disabled by default for dialects that don't allow double quotes for
       quoted literals (e.g. Postgres). It can be enabled with the
       ``force_enable = True`` flag.
```